client/post-editor/editor-revisions/index.jsx

```diff
@@ -1,53 +1,68 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
 import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import { partial } from 'lodash';
+import React, { PropTypes } from 'react';
 
-export default React.createClass( {
-
-	displayName: 'EditorRevisions',
-
-	mixins: [ PureRenderMixin ],
-
-	propTypes: {
-		adminUrl: React.PropTypes.string,
-		revisions: React.PropTypes.array
-	},
-
-	getDefaultProps() {
-		return {
-			revisions: []
-		};
-	},
-
-	render() {
-		if ( ! this.props.revisions || ! this.props.revisions.length ) {
-			return null;
-		}
+/**
+ * Internal dependencies
+ */
+import { isEnabled } from 'config';
+import { NESTED_SIDEBAR_REVISIONS } from 'post-editor/editor-sidebar/util';
 
-		const lastRevision = this.props.revisions[ 0 ];
-		const revisionsLink = this.props.adminUrl + 'revision.php?revision=' + lastRevision;
+function EditorRevisions( { adminUrl, revisions = [], translate, toggleNestedSidebar } ) {
+	if ( ! revisions || ! revisions.length ) {
+		return null;
+	}
 
+	if ( isEnabled( 'post-editor/revisions' ) ) {
 		return (
-			<a className="editor-revisions"
-				href={ revisionsLink }
-				target="_blank"
-				rel="noopener noreferrer"
-				aria-label={ this.translate( 'Open list of revisions' ) }
+			<button
+				className="editor-revisions"
+				title={ translate( 'Open list of revisions' ) }
+				onClick={ partial( toggleNestedSidebar, NESTED_SIDEBAR_REVISIONS ) }
```

Comment:
Instantiating functions within `render` is an anti-pattern. Actually, ESLint will complain if it sees an actual `() =>` construct in here, but it is blind to `partial`. :)